doc/source/whatsnew/v3.0.0.rst

```diff
@@ -325,6 +325,7 @@ Bug fixes
 - Fixed bug in :class:`SparseDtype` for equal comparison with na fill value. (:issue:`54770`)
 - Fixed bug in :meth:`.DataFrameGroupBy.median` where nat values gave an incorrect result. (:issue:`57926`)
 - Fixed bug in :meth:`DataFrame.cumsum` which was raising ``IndexError`` if dtype is ``timedelta64[ns]`` (:issue:`57956`)
+- Fixed bug in :meth:`DataFrame.eval` and :meth:`DataFrame.query` which caused an exception when using numpy via ``@`` notation. (:issue:`58041`)
```

Comment:
I'm not sure if "when using numpy" is clear - what do you think of doing something like

> when using NumPy attributes via ``@`` notation, e.g. ``df.eval("@np.floor(a)")``

Reply:
Updated as per suggestion, thanks